### PR TITLE
Add a PrepareError registry

### LINKED ISSUE #487: Registry for PrepareError
[draft-priebe-ppm-dap-reportauth](https://github.com/cpriebe/draft-priebe-ppm-dap-reportauth) (no I-D submitted yet) wants to define a new `PrepareError` type, but we do not have a registry for this, only the enum definition. In conjunction with setting up the necessary policy for `PrepareError` extensibility, we may want to expand the encoded size from one byte to two bytes. We are already using 4% of codepoints for DAP itself, and additional headroom would make managing this namespace easier.

### PR DESCRIPTION
**Pull request by chris-wood**
Closes #487

### CODE REVIEW

File: draft-ietf-ppm-dap.md [hunk 1]
```diff
@@ -1404,7 +1404,7 @@ enum {
   task_expired(7),
   invalid_message(8),
   report_too_early(9),
-  (255)
+  (65535)
```
Comment (tgeoghegan) [inline draft-ietf-ppm-dap.md:1407]: Can you explain the choice to widen this error type? Are we ever going to define this many distinct error codes? Admittedly errors should be the minority of messages so this extra byte won't cost extra bandwidth in the common case of the protocol.

Comment (cjpatton) [inline draft-ietf-ppm-dap.md:1407]: Why bump 1-byte to 2-byte?

**Comment by chris-wood** [comment]
@tgeoghegan @cjpatton the size increase is to account for random selection of new error types for experimentation. But we can do that separately, if at all, so I'll revert it from this PR.

File: draft-ietf-ppm-dap.md (hunk 1 above)
Comment (chris-wood) [inline draft-ietf-ppm-dap.md:1407]:
```suggestion
  (255)
```

File: draft-ietf-ppm-dap.md [hunk 2]
```diff
@@ -3414,20 +3414,43 @@ Change controller:
...
+### Prepare Error Registry {#prepare-error-reg}
+
+This document requests creation of a new registry for PrepareError values. This
+registry should contain the following columns:
+
+Name:
+: The name of the PrepareError value
+
+Value:
+: The 2-byte value of the PrepareError value
```
Comment (tgeoghegan) [inline draft-ietf-ppm-dap.md:3446]:
```suggestion
: The 1-byte value of the PrepareError value
```

**Review by tgeoghegan — APPROVED** [review]

**Review by cjpatton — APPROVED** [review]

**Comment by cjpatton** [comment]
@chris-wood can you squash?

**Comment by cjpatton** [comment]
Rebased and squashed.

**Pull request merged**